components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/grant/DeviceFlowGrant.java

```diff
@@ -64,32 +65,61 @@ public boolean validateGrant(OAuthTokenReqMessageContext oAuthTokenReqMessageCon
             log.debug("Getting ready to release token for device_code: " + deviceCode);
         }
 
-        DeviceFlowDO deviceFlowDO = DeviceFlowPersistenceFactory.getInstance().getDeviceFlowDAO()
-                .getAuthenticationDetails(deviceCode, clientId);
-        Date date = new Date();
-        deviceStatus = deviceFlowDO.getStatus();
-        deviceFlowDO.setDeviceCode(deviceCode);
-        if (Constants.NOT_EXIST.equals(deviceStatus)) {
-            throw new IdentityOAuth2Exception(DeviceErrorCodes.INVALID_REQUEST, DeviceErrorCodes.INVALID_REQUEST);
+        try {
+            deviceFlowDO = DeviceFlowPersistenceFactory.getInstance().getDeviceFlowDAO()
+                    .getAuthenticationDetails(deviceCode, clientId);
+            deviceStatus = deviceFlowDO.getStatus();
+            deviceFlowDO.setDeviceCode(deviceCode);
+            setLastPollTime(deviceCode);
+        } catch (IdentityOAuth2Exception e) {
+            deviceStatus = e.getMessage();
```

Comment:
we don't need this line right?

Reply:
As we need to pass it here, https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2587/files#diff-04c8d9d96bce4315cd10faf6cee77e67804ef26f765ac63eda3f36ab5767b945R80 we need to specify here.

Reply:
this catch block don't get execute after line 76,  cause ``` handleInvalidRequests ``` method throws identityOAuthException anyway.